### PR TITLE
Retry requests to marketdata API that fail due to rate limiting

### PR DESCRIPTION
**Pull request by gpoul**
I tested with my downloaded file from Wiener Netze that included the last year of detail data and ran into the issue that it failed to load and just produced errors in the browser console; I've added some code that retries those failed requests.

Main issue is that Awattar marketdata API is not adding CORS headers to those requests, so we cannot do nice retries and I think we have to solve this with try/catch blocks "guessing" the error that happened.

### CODE REVIEW

File: docs/app.js [hunk 1]
```diff
@@ -107,6 +107,32 @@ class Awattar {
...
+    var retryFetch = 0;
+    do {
+        var response;
+        try {
+            response = await fetch('https://api.awattar.at/v1/marketdata?start=' + unixStamp)
+        } catch (error) {
+            console.log("Requested failed; will retry to get Awattar market data:", error);
+            retryFetch++;
+        }
+        if (response && response.ok || retryFetch > 10) {
```
Comment (lewurm) [inline docs/app.js:123]: please add parentheses

**Comment by lewurm** [comment]
Thanks!  But I'm surprised you had problems with it.  Although aWATTar states the following on their website:

> Nutzen Sie unseren Datenfeed kostenlos nach dem Fair Use Prinzip (100 Abfragen pro Tag).

I haven't seen this enforced, but I'm wondering if a sleep would help then.  Are you sure your PR actually fixes that?  Note that there is a cache in the `Local Storage` of the browser (see key `awattarCache`), to avoid spamming the API.  For testing your PR however you need to delete that cache first.

**Comment by gpoul** [comment]
I can document you what I'm seeing in the requests later this week. The cache only works when all API requests are successful. They're not persisted incrementally, so my full-year file never causes a cache population in the original version. 

I'm not sure what Awattar does, but it sends a header to back off for 6s when the first  unsuccessful API call is processed. With 6s wait I only got two more requests and then it throttled again. With the 10s backoff it seemed fine. 

They also don't send CORS headers in the failures so that's also not useful and I can't seem to get to the request headers then. 

I think I should be able to produce a full-year test file that shows the same error behavior if you think that might be useful.

**Comment by lewurm** [comment]
Thanks @gpoul!

> I think I should be able to produce a full-year test file that shows the same error behavior if you think that might be useful.

If it isn't too much effort that would be indeed useful. Please add it to the `sample/` folder.

**Comment by gpoul** [comment]
I added the test files to the `samples/` folder and also checked that I still get the same error with that

**Comment by lewurm** [comment]
> I added the test files to the `samples/` folder and also checked that I still get the same error with that

Thanks!  I could reproduce it when I delete my local storage first.  Thanks for the fix 👍

**Comment by lewurm** [comment]
FYI apparently aWATTar got stricter.  Technically your workaround still works, but it takes waaaaaay longer now.

So I went ahead and did something else: I basically mirror each day of the last four years on this site https://github.com/awattar-backtesting/awattar-backtesting.github.io/commit/368356de6e3ad8dc1b898952244e0501f8e97d3d

Not great, but it does the job 😅

**Pull request merged**